android/engine/src/main/java/org/smartregister/fhircore/engine/task/WorkflowCarePlanGenerator.kt

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright 2021-2023 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.task
+
+import android.content.Context
+import ca.uhn.fhir.context.FhirContext
+import ca.uhn.fhir.context.FhirVersionEnum
+import com.google.android.fhir.FhirEngine
+import com.google.android.fhir.knowledge.KnowledgeManager
+import com.google.android.fhir.search.search
+import com.google.android.fhir.workflow.FhirOperator
+import dagger.hilt.android.qualifiers.ApplicationContext
+import java.io.File
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.hl7.fhir.r4.model.Bundle
+import org.hl7.fhir.r4.model.CanonicalType
+import org.hl7.fhir.r4.model.CarePlan
+import org.hl7.fhir.r4.model.IdType
+import org.hl7.fhir.r4.model.Library
+import org.hl7.fhir.r4.model.MetadataResource
+import org.hl7.fhir.r4.model.Patient
+import org.hl7.fhir.r4.model.PlanDefinition
+import org.hl7.fhir.r4.model.Reference
+import org.hl7.fhir.r4.model.Resource
+import org.hl7.fhir.r4.model.ResourceType
+import org.hl7.fhir.r4.model.Task
+
+@Singleton
+class WorkflowCarePlanGenerator
+@Inject
+constructor(
+  val knowledgeManager: KnowledgeManager,
+  val fhirOperator: FhirOperator,
+  val fhirEngine: FhirEngine,
```

Comment:
think we need to replace FhirEngine w/DefaultRepository and change all the calls to fhirEngine to call against defaultRepo

Reply:
updated `fhirEngine` with `defaultRepository`